RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptions/ManageSubscriptionsViewModel.swift

```diff
@@ -263,10 +263,16 @@ private extension Array<CustomerCenterConfigData.HelpPath> {
             return self
         }
 
-        return filter { !purchaseInformation.isLifetime || $0.type != .cancel }
-            .filter {
-                $0.refundWindowDuration.map { $0.isWithin(purchaseInformation) } ?? true
-            }
+        return filter {
+            // if it's cancel, it cannot be a lifetime subscription
+            ($0.type != .cancel || !purchaseInformation.isLifetime) &&
+
+            // if it's refundRequest, it cannot be free
+            ($0.type != .refundRequest || purchaseInformation.price != .free) &&
+
+            // if it has a refundDuration, check it's still valid
+            ($0.refundWindowDuration?.isWithin(purchaseInformation) ?? true)
+        }
```

Comment:
Thanks for adding the comments. It's also much clearer checking the type first. I would have done positive checks instead of negative conditions, but it's a personal preference

Reply:
I think it might be a good idea to tweak it if this keeps growing. I'll keep it in mind :)